instrumentation/opentelemetry-instrumentation-asyncpg/src/opentelemetry/instrumentation/asyncpg/__init__.py

```diff
@@ -98,6 +104,9 @@ def __init__(self, capture_parameters=False):
         self.capture_parameters = capture_parameters
         self._tracer = None
 
+    def instrumentation_dependencies(self) -> Collection[str]:
```

Comment:
Can this be part of `BaseInstrumentor`?

Reply:
Possibly but I think it'd be hacky. BaseInstrumentor needs to know it's sub-class and from there try to dynamically import/load information about dependencies. In any case, BaseInstrumentor will need to figure out the final implementation and also have a way to load/extract data about dependencies from it. Even if we use some sort of convention over configuration and some runtime magic to figure things out, I think it'd put too much burder and complexity into the BaseInstrumentor just to save a couple of lines on each final implementation. 

Happy to hear other ideas though.

Reply:
Yeah I think the hackiness would be worse than the duplicated code so I think it is fine to leave it as is.